nbbuild/templates/common.xml

```diff
@@ -78,10 +78,10 @@
         <tstamp>
             <format property="module.build.started.time" pattern="yyyy-MM-dd'T'HH:mm:ss.SSSZ"/>
         </tstamp>
-        <condition property="asm.jar" value="${nbplatform.active.dir}/platform/core/asm-all-5.0.1.jar">
-            <available file="${nbplatform.active.dir}/platform/core/asm-all-5.0.1.jar"/>
+        <condition property="asm.jar" value="${nbplatform.active.dir}/platform/core/asm-7.2.jar">
```

Comment:
This `${asm}` is passed as asm location to `processjsannotations` task, but its property only accepts a single file (used to be `asm-all*.jar`) so the `-tree` part can be now missing from task's runtime.
@JaroslavTulach wrote the task

Reply:
Both the asm-7.2. & asm-tree-7.2 jars appears in the Startup Classpath. Or is this a different topic?
[asm-72v7-dedic-patch.log](https://github.com/apache/netbeans/files/3846985/asm-72v7-dedic-patch.log)

Reply:
See [line 209](https://github.com/apache/netbeans/blob/5da8a26a2435256f969d7ae3f49a2964bfc6b361/nbbuild/templates/common.xml#L209): 
```
        <processjsannotation classes="${build.classes.dir}" asm="${asm.jar}">
            <classpath refid="cp"/>
        </processjsannotation>
```
The task first loads asm from `cp`, then from the `asm` property; I don't know if it is guaranteed that both ASM jars appear on `cp` (maybe not, because that's why the task has an alternative property). But that property only allows to add one jar, we have two of them now. @jtulach  should know better what that task requires.

Reply:
I don't remember clearly, but I think that inside of `nbbuild` one classpath is valid and inside of `harness` build another classpath is valid.

Reply:
I am not able to find an example to test with. The `FnUtils` class that is reflective-loaded by `processjsannotations` is at https://github.com/apache/netbeans-html4j/blob/master/boot/src/main/java/org/netbeans/html/boot/impl/FnUtils.java ; looking at the source, it should be satisfied with just the basic `asm`, seems it does not need types from `asm-tree`.

Reply:
Doing a "Find in Project" of `org.objectweb.asm.tree` just three files use `asm-tree`

- `ide/ide.kit/test/qa-functional/data/whitelist_1.txt`
- `java/lib.jshell.agent/agentsrc/org/netbeans/lib/jshell/agent/NbJShellAgent.java`
- `platform/core.startup/src/org/netbeans/core/startup/Asm.java`

Reply:
Do you think the 'asm' parameter to the `processjsannotations` task is still needed? Could the relevant jars be included as an element in the 'classpath' parameter? In other words, do the asm jars need to be treated specially? As sdedic says, the logic now is that the contents of the asm parameter are not used if anything in the classpath parameter contains the letters "asm", and I don't know enough of the history to understand why that might be.